### PR TITLE
⚙️ [pi-harness] feat(bridge): register Pi and OMP [step 18/21]

### PR DESCRIPTION
**Pull request by AlexDochioiu**
## Complexity

⚙️ Moderate: additive registration across shared enum, bridge registry, and a new client toast surface; no wire, database, or lifecycle machinery changes.

## What

- Add `Harness.pi` and `Harness.omp` to the shared built-in harness enum.
- Add `pi_plugin`/`omp_plugin` dependencies to the bridge app and register `PiPluginDescriptor.production()` and `OmpPluginDescriptor.production()` in `plugin_registry.dart` (list became `final List.unmodifiable` — both descriptors use non-const factories). OpenCode stays the preferred default; existing entries preserved.
- Add backend-neutral `SseToastCubit` in `module_core`: maps `tui.toast.show` SSE events to sealed idle/show states with a monotonic sequence so equal repeated guidance re-fires; textless toasts drop; unknown variants degrade to info. The mobile shell renders shows as snackbars via a new root scaffold messenger.
- Record mandatory activation guidance in README and the plugin-setup regression doc: Pi always runs with `--approve` (project-local Pi settings/extensions/skills/templates trusted without prompts), OMP inherits the user's `tools.approvalMode`, and provider login is local-only.
- Update registry exact-set fixtures and add a test pinning every registered id to a built-in `Harness` identity.

## Why

Step 18 of the pi-harness plan: both backend plugins are complete (Steps 16-17 merged), so the bridge can register them and the app can select them. The toast surface is the minimum safety presentation required before activation — backend notifications and local `/login` guidance must reach the user.

## Risk and test focus

Low-moderate risk. Registration is additive and inert (descriptors are side-effect free); the main flows to watch are harness listing/default selection and the new global toast rendering. Most valuable checks: registry exact set, preferred default unchanged, toast monotonic repeat/empty-drop/variant fallback, and that older clients treat the new `pi`/`omp` ids through existing data-driven fallback (generic icon, raw-id name).

## Expected result

Pi and Oh My Pi appear as selectable harnesses on the bridge and phone once their setup is ready, with their trust/approval guidance documented, and backend toast guidance reaches users on any screen. No database or persisted-data change; wire contracts unchanged (plugin ids remain strings).

## Verification

- Bridge app: `dart analyze --fatal-infos` clean; full suite 2,674 tests pass.
- Pi plugin: 255 tests; OMP plugin: 51 tests; shared: 409 tests.
- module_core: toast cubit tests pass; analyzer clean. Mobile shell: `flutter analyze --fatal-infos` clean.
- Architecture implementation review: approved, no findings.

## Diff evidence

- +393/−10 = 403 changed lines before plan/tracker evidence; generated lines: 46.

---
## Summary by cubic
Registers Pi and Oh My Pi as built-in harnesses and routes backend toast guidance through design‑system popups that respect active top bars/banners. Previously Pi/OMP were not selectable and toasts were inconsistent or obscured; now both appear in listings and tui.toast.show renders app‑wide below banners on any screen.

- Bridge: add Harness.pi/Harness.omp; register `PiPluginDescriptor.production()` and `OmpPluginDescriptor.production()`; switch `knownPlugins` to `List.unmodifiable`; add `pi_plugin` and `omp_plugin`. OpenCode remains the preferred default.
- Toasts: add SseToastCubit mapping `tui.toast.show` SSE to idle/show with a monotonic sequence; drop textless toasts; normalize blank titles; unknown variants degrade to info. Render with PregoPopupAlertPresenter on the root navigator’s overlay (exposes appRootNavigatorKey and adds fromOverlayState). PregoGlassScaffold publishes live root top‑bar inset per overlay so global popups sit below active banners and restore the underlying route after a pop; publisher identities are now typed and kept out of the `module_prego` barrel. The old root‑ScaffoldMessenger path and splash scaffolding are not needed.
- Docs: document Pi’s always‑`--approve` trust, OMP’s inherited `tools.approvalMode`, and local‑only provider login in README and regression docs.
- Tests/compat: pin the registry exact set (includes `pi`/`omp`) and map each id to a built‑in Harness; add cubit tests for repeat, fallback, title normalization, and empty‑drop; add overlay/inset presenter tests. No database or wire changes; descriptors remain inert; older clients handle new ids via existing fallback.

<sup>Written for commit 2efa57327958355733adac33367ba53ab44b7e53. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/sesori-ai/sesori_apps_monorepo/pull/967?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: client/app/lib/main.dart [hunk 1]
```diff
@@ -336,10 +341,40 @@ class const _SesoriAppShell() extends StatelessWidget {
...
+class const _SseToastListener({required final Widget child}) extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return BlocListener<SseToastCubit, SseToastState>(
+      listener: (context, state) {
+        if (state case SseToastShow(:final title, :final message, :final variant)) {
+          final messenger = rootScaffoldMessengerKey.currentState;
+          if (messenger == null) return;
+          final text = title == null ? message : "$title\n$message";
+          messenger.showSnackBar(
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply client/app/lib/main.dart]: Fixed in `b82197ddb`: the splash view now provides a `Scaffold`, so the root messenger always has a descendant during startup and a toast arriving on the splash route presents instead of asserting.

File: client/module_core/lib/src/cubits/sse_toast/sse_toast_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    if (event.data case SesoriTuiToastShow(:final title, :final message, :final variant)) {
+      final trimmedTitle = title?.trim();
+      final trimmedMessage = message?.trim();
+      final text = trimmedMessage == null || trimmedMessage.isEmpty ? trimmedTitle : trimmedMessage;
+      if (text == null || text.isEmpty) return;
+      if (isClosed) return;
+      emit(
+        SseToastState.show(
+          sequence: ++_sequence,
+          title: trimmedMessage == null || trimmedMessage.isEmpty ? null : trimmedTitle,
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply client/module_core/lib/src/cubits/sse_toast/sse_toast_cubit.dart]: Fixed in `b82197ddb`: a whitespace-only title now normalizes to `null` at the source, so no leading blank line reaches the listener. Covered by a new cubit test.

File: client/app/lib/main.dart [hunk 3]
```diff
@@ -336,10 +341,40 @@ class const _SesoriAppShell() extends StatelessWidget {
...
+/// guidance such as a local `/login` hint reaches the user on any screen.
+class const _SseToastListener({required final Widget child}) extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return BlocListener<SseToastCubit, SseToastState>(
+      listener: (context, state) {
+        if (state case SseToastShow(:final title, :final message, :final variant)) {
+          final messenger = rootScaffoldMessengerKey.currentState;
+          if (messenger == null) return;
+          final text = title == null ? message : "$title\n$message";
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply client/app/lib/main.dart]: Addressed at the source in `b82197ddb`: `SseToastCubit` normalizes a blank title to `null`, which makes the blank-title-with-message combination unrepresentable here, so the listener's `title == null` check now suffices.

File: .plan/active/pi-harness/TRACKER.md [hunk 4]
```diff
@@ -562,6 +562,34 @@ binary install. `OMP_PROTOCOL.md` records the supporting evidence.
...
+- Registry exact-set fixtures updated; a new test pins every registered id to
+  a built-in `Harness` identity. Toast cubit tests cover monotonic repeats,
+  variant fallback, and empty-drop behavior.
+- Architecture implementation review approved the commit with no findings.
+- Verification: bridge app fatal analysis and full suite (2,674 tests), Pi
+  (255) and OMP (51) suites, shared (409), module_core toast tests, and the
+  mobile shell fatal analysis all pass after the post-merge sync with `main`.
+- No database, persisted-data, or wire-contract change; the app now lists Pi
+  and Oh My Pi as selectable harnesses once their setup is ready.
+- Diff before tracker/plan evidence: +393/-10 = 403 changed lines; generated
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply .plan/active/pi-harness/TRACKER.md:591]: The 800–1,200 line budget in the plan was an over-estimate for this step: the registration itself is small (an enum case, two descriptor entries, dependency lines), the toast cubit and rendering landed compactly, and no Pi/OMP client-specific checks were needed since existing attachment capability already drives composer behavior. The measured ~450–500 lines (including tracker evidence) is the honest total; the plan budgets are scoping aids, not targets.

**Comment by AlexDochioiu** [comment]
Toast presentation switched to the design-system surface in `7784d42a8`: backend `tui.toast.show` guidance now renders through `PregoPopupAlertPresenter` on the root navigator's overlay (same component used for "Session deleted" and friends) instead of a plain root-messenger `SnackBar`. The splash `Scaffold` workaround is reverted since the overlay path presents during startup without one, and the presenter gained a `fromOverlayState` factory for callers without a route context.

File: client/app/lib/main.dart [hunk 5]
```diff
@@ -336,10 +336,45 @@ class const _SesoriAppShell() extends StatelessWidget {
...
+/// Renders backend toast states through the design-system popup alert
+/// presenter, so guidance such as a local `/login` hint reaches the user on
+/// any screen, including startup routes with no scaffold.
+class const _SseToastListener({required final Widget child}) extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return BlocListener<SseToastCubit, SseToastState>(
+      listener: (context, state) {
+        if (state case SseToastShow(:final title, :final message, :final variant)) {
+          final overlay = appRootNavigatorKey.currentState?.overlay;
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply client/app/lib/main.dart:359]: Valid and fixed in `5c27f8577`. `PregoGlassScaffold` now publishes its live base + animated banner height through a root seam keyed by the exact `OverlayState`; `PregoPopupAlertPresenter.fromOverlayState` reads only that overlay’s active geometry. Mounted ordering prevents covered routes from displacing the topmost route during shared rebuilds and restores the underlying route after pop. Added focused tests for explicit-overlay banner clearance and mounted ordering; module_prego analyze/full 209 tests and app analyze pass. Architecture re-review approved the overlay-scoped seam.
